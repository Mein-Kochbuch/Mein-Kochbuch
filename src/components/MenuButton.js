import React from "react";
import {StyleSheet, Text, TouchableOpacity} from "react-native";
import globalStyles from "../styles/globalStyles";
import {useHistory} from "react-router-native";

export default function MenuButton({text, style, path}) {
    const {border} = globalStyles()
    const history = useHistory()

    const styles = StyleSheet.create({
        container: {
            display: "flex",
            flexDirection: "column",
            justifyContent: "space-between",
            alignItems: "center",
            width: "80%",
            minWidth: 300,
        },
        text: {
            textAlign: "center",
            fontSize: 30,
        }
    })

    const onPress = (to) => {
        history.push(`/${to}`)
    }

    return (
        <TouchableOpacity onPress={() => onPress(path)} style={{...border, ...styles.container, ...style}}>
            <Text style={styles.text}>
                {text}
            </Text>
        </TouchableOpacity>
    )
}
