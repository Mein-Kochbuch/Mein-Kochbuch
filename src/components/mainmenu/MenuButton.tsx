import React from "react";
import {useHistory} from "react-router-native";
import styled from "styled-components/native";
import {useColorScheme} from "react-native-appearance";
import {Colors} from "react-native/Libraries/NewAppScreen";

interface MenuButtonProps {
    text: string,
    path: string
}

export default function MenuButton({text, path}: MenuButtonProps) {
    const history = useHistory()
    const isDarkMode = useColorScheme() === 'dark';

    const onPress = (to: string) => {
        history.push(`/${to}`)
    }

    return (
        <StyledTouchableOpacity isDarkMode={isDarkMode} onPress={() => onPress(path)}>
            <StyledText>
                {text}
            </StyledText>
        </StyledTouchableOpacity>
    )
}

const StyledText = styled.Text`
  text-align: center;
  font-size: 30px;
`

interface StyledTouchableOpacityProps {
    isDarkMode: boolean
}

const StyledTouchableOpacity = styled.TouchableOpacity<StyledTouchableOpacityProps>`
  margin: 12px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  width: 80%;
  min-width: 300px;
  border-style: solid;
  border-color: ${props => props.isDarkMode ? Colors.lighter : Colors.darker};
  border-radius: 10px;
  border-width: 2px;
`
