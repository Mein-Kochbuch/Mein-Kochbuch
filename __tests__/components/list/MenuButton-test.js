import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import MenuButton from '../../../src/components/mainmenu/MenuButton';
import {fireEvent, render} from '@testing-library/react-native';
import {Router} from 'react-router-native';

describe('MenuButton Test', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('MenuButton Button Press', () => {
    const history = {
      push: jest.fn(),
      listen: jest.fn(),
      location: {pathname: '/'},
    };

    const {getByText} = render(
      <Router history={history}>
        <MenuButton text={'Test-title'} path={'test'} />
      </Router>,
    );

    fireEvent(getByText('Test-title'), 'onPress');
    expect(history.push).toBeCalledWith('/test');
  });

  it('MenuButton lightmode Test', () => {
    const component = renderer
      .create(<MenuButton text={'Test-text'} />)
      .toJSON();

    expect(component).toMatchSnapshot();
    expect(component.children.length).toBe(1);
    expect(component.children[0].children[0]).toBe('Test-text');
  });
});
