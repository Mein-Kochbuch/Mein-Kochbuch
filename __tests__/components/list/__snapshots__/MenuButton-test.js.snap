// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuButton Test MenuButton lightmode Test 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#222",
      "borderRadius": 10,
      "borderStyle": "solid",
      "borderWidth": 2,
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "space-between",
      "marginBottom": 12,
      "marginLeft": 12,
      "marginRight": 12,
      "marginTop": 12,
      "minWidth": 300,
      "opacity": 1,
      "width": "80%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 30,
          "textAlign": "center",
        },
      ]
    }
  >
    Test-text
  </Text>
</View>
`;
